docs/technology_models/grid.md

```diff
@@ -0,0 +1,70 @@
+# Grid Performance and Cost Models
+
+This page documents the unified `grid_performance` and `grid_cost` models, which together represent a flexible, configurable grid interconnection point within an H2I simulation.
+These components support both power flows and cost accounting for buying and selling electricity through a constrained interconnection.
+This is a single model that can be configured to either sell electricity to the grid, buy electricity from the grid, or both.
+
+See `example/24_solar_battery_grid` to see how to set up both buying and selling grid components.
+
+## Grid Performance
+`grid_performance` represents a grid interconnection point that can buy or sell electricity subject to a maximum throughput rating (interconnection_size).
+
+It supports:
+- Buying electricity from the grid to meet downstream demand.
+- Selling electricity to the grid.
+- Enforcing maximum allowed interconnection power.
+- Computing unmet demand and unsold electricity due to constraints.
+
+```{note}
+Multiple grid instances may be used within the same plant to represent different interconnection nodes. In the `tech_config` the component **must** be called `grid_buy` for the logic to work appropriately in financial calculations.
+```
+
+**Inputs**
+| Name                     | Shape              | Units | Description                                                       |
+| ------------------------ | ------------------ | ----- | ----------------------------------------------------------------- |
+| `interconnection_size`   | scalar             | kW    | Maximum power capacity for grid connection.                       |
+| `electricity_in`         | array[n_timesteps] | kW    | Electricity flowing into the grid (selling to grid).              |
+| `electricity_demand`     | array[n_timesteps] | kW    | Electricity demand from downstream technologies.                  |
+
+**Outputs**
+| Name                       | Shape              | Units | Description                                       |
+| -------------------------- | ------------------ | ----- | ------------------------------------------------- |
+| `electricity_out`          | array[n_timesteps] | kW    | Electricity flowing *out of* the grid (buying).   |
+| `electricity_sold`         | array[n_timesteps] | kW    | Electricity successfully sold to the grid.        |
+| `electricity_unmet_demand` | array[n_timesteps] | kW    | Demand not met due to interconnection limits.     |
+| `electricity_not_sold`     | array[n_timesteps] | kW    | Electricity that could not be sold due to limits. |
+
+## Grid Cost
+`grid_cost` computes all costs associated with the grid interconnection, including:
+- Capital cost based on interconnection rating.
+- Fixed annual O&M.
+- Variable cost of electricity purchased.
+- Revenue from electricity sold.
+- Optional time-varying energy prices.
```

Comment:
The wording here is a little weird. It makes it sound like these are outputs of the model, but time-varying energy prices would be an input?